lib/kubernetes-deploy/resource_watcher.rb

```diff
@@ -41,14 +41,19 @@ def run(delay_sync: 3.seconds, reminder_interval: 30.seconds)
     def report_what_just_happened(resources, watch_time)
       new_successes, new_failures = resources.partition(&:deploy_succeeded?)
       new_failures.each do |resource|
+        tags = %W(namespace:#{@namespace} sha:#{ENV['REVISION']} resource:#{resource.id})
```

Comment:
> We can add the type if we want, though just having the failed name is should be enough

`id` is in the form `type/name`, so this is emitting both, but tacked together. My thinking is that if you emit those separately, you'd also be able to do things like see things like one type in general taking longer than others--e.g. Pods, which always have unique ids between deploys. Otoh names aren't necessarily unique, so perhaps the useful combination would be `id` and `type` as you said. I'm just speculating though.